src/React.Core/IReactSiteConfiguration.cs

```diff
@@ -43,17 +43,48 @@ public interface IReactSiteConfiguration
 		/// <returns>The configuration, for chaining</returns>
 		IReactSiteConfiguration AddScriptWithoutTransform(string filename);
 
-		/// <summary>
-		/// Gets a list of all the scripts that have been added to this configuration and require JSX
-		/// transformation to be run.
-		/// </summary>
-		IEnumerable<string> Scripts { get; }
+        /// <summary>
+        /// Adds a script literal to the list of scripts that are executed. If the script doesn't have any JSX in it
+        /// use <see cref="AddScriptLiteralWithoutTransform"/> instead.
+        /// </summary>
+        /// <param name="scriptId">The id of the script, used for caching purposes.</param>
+        /// <param name="script">The script literal to execute.</param>
+        /// <returns>The configuration, for chaining</returns>
+        IReactSiteConfiguration AddScriptLiteral(string scriptId, string script);
 
-		/// <summary>
-		/// Gets a list of all the scripts that have been added to this configuration and do not 
-		/// require JSX transformation to be run.
+        /// <summary>
+        /// Adds a script literal to the list of scripts that are executed. This is the same as
+        /// <see cref="AddScriptLiteral"/> except it does not run JSX transformation on the script and thus is
+        /// more efficient.
+        /// </summary>
+        /// <param name="script">
+        /// The script literal to execute.
+        /// </param>
+        /// <returns>The configuration, for chaining</returns>
+        IReactSiteConfiguration AddScriptLiteralWithoutTransform(string script);
```

Comment:
Do you think this is useful? It seems like overkill :) How about just adding `AddScriptLiteral` for now?